File: library/src/androidTest/mock/com/mobilesolutionworks/android/app/test/basic/TestWorksDialogFragment.java
package com.mobilesolutionworks.android.app.test.basic;

import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v7.app.AlertDialog;

import com.mobilesolutionworks.android.app.WorksDialogFragment;
import com.mobilesolutionworks.android.app.controller.WorksControllerManager;
import com.mobilesolutionworks.android.app.test.GetController;

/**
 * Created by yunarta on 9/3/17.
 */

public class TestWorksDialogFragment extends WorksDialogFragment implements GetController<TestWorksController> {

    private TestWorksController mController;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mController = getControllerManager().initController(0, null, new WorksControllerManager.ControllerCallbacks<TestWorksController>() {
            @Override
            public TestWorksController onCreateController(int id, Bundle args) {
                return new TestWorksController();
            }
        });
    }

    public TestWorksController getController() {
        return mController;
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        return new AlertDialog.Builder(getContext())
                .setMessage("Example")
                .setPositiveButton("CLOSE", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        dismiss();
                    }
                })
                .create();
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        mController = null;
    }
}

File: test-app/src/main/kotlin/com/mobilesolutionworks/android/controller/test/basic/TestWorksDialogFragment.kt
package com.mobilesolutionworks.android.controller.test.basic

import android.app.Dialog
import android.os.Bundle
import android.support.v7.app.AlertDialog
import com.mobilesolutionworks.android.app.WorksDialogFragment
import com.mobilesolutionworks.android.app.controller.CreateCallback2
import com.mobilesolutionworks.android.controller.test.GetController

/**
 * Created by yunarta on 9/3/17.
 */

class TestWorksDialogFragment : WorksDialogFragment(), GetController<TestWorksController> {

    override var controller: TestWorksController? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        controller = controllerManager.initController(0, null, CreateCallback2 { TestWorksController(it) })
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        return AlertDialog.Builder(context!!)
              .setMessage("Example")
              .setPositiveButton("CLOSE") { dialog, which -> dismiss() }
              .create()
    }

    override fun onDestroy() {
        super.onDestroy()
        controller = null
    }
}
